src/envoy/http/metadata_exchange/config.cc

```diff
@@ -23,36 +23,49 @@ namespace MetadataExchange {
 
 // imports from the low-level API
 using Common::Wasm::Null::NullVmPluginFactory;
+using Common::Wasm::Null::Plugin::getMetadataStringValue;
 using Common::Wasm::Null::Plugin::getMetadataStruct;
 using Common::Wasm::Null::Plugin::getMetadataValue;
 using Common::Wasm::Null::Plugin::getRequestHeader;
 using Common::Wasm::Null::Plugin::getResponseHeader;
 using Common::Wasm::Null::Plugin::logDebug;
 using Common::Wasm::Null::Plugin::logInfo;
+using Common::Wasm::Null::Plugin::logWarn;
 using Common::Wasm::Null::Plugin::removeRequestHeader;
 using Common::Wasm::Null::Plugin::removeResponseHeader;
 using Common::Wasm::Null::Plugin::replaceRequestHeader;
 using Common::Wasm::Null::Plugin::replaceResponseHeader;
+using Common::Wasm::Null::Plugin::setMetadataStringValue;
+
+bool serializeToStringDeterministic(const google::protobuf::Struct& metadata,
+                                    std::string* metadata_bytes) {
+  google::protobuf::io::StringOutputStream md(metadata_bytes);
+  google::protobuf::io::CodedOutputStream mcs(&md);
+
+  mcs.SetSerializationDeterministic(true);
+  if (!metadata.SerializeToCodedStream(&mcs)) {
+    logWarn(
+        absl::StrCat("unable to serialize Nodemetadata key=", NodeMetadataKey));
+    return false;
+  }
```

Comment:
I could get rid of `mcs.Trim()` by putting this in a separate function.
Now `CodedOutputStream` is destroyed before the `std::string` is used by the caller, this ensures that flush/Trim is called.

Reply:
nit: I would suggest return `absl::optional<string>()`
I think we can leave it in future versions as this PR does nothing wrong :)